device/cuda/CMakeLists.txt

```diff
@@ -46,7 +46,7 @@ traccc_add_library( traccc_cuda cuda TYPE SHARED
   "src/fitting/fitting_algorithm.cu")
   
 target_compile_options( traccc_cuda
-  PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:--expt-relaxed-constexpr> )
+  PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:--expt-relaxed-constexpr -Xcompiler -finstrument-functions -fPIC -lnvToolsExt> )
```

Comment:
I would rather not inject this directly into every build. Perhaps we can inject this into `RelWithDebInfo` bugs, although I am also not very fond of introducing performance effects there that do not exist in `Release` builds. Let's add a `Profile` build type or something similar.